include/boost/geometry/algorithms/detail/disjoint/linear_areal.hpp

```diff
@@ -65,15 +66,29 @@ struct disjoint_no_intersections_policy
     /*!
     \tparam Strategy point_in_geometry strategy
     */
+    typedef typename point_type<Geometry1>::type point1_type;
+
     template <typename Strategy>
-    static inline bool apply(Geometry1 const& g1, Geometry2 const& g2, Strategy const& strategy)
+    static inline bool apply(Geometry1 const& g1,
+                             Geometry2 const& g2,
+                             Strategy const& strategy)
     {
-        typedef typename point_type<Geometry1>::type point1_type;
         point1_type p;
         geometry::point_on_border(p, g1);
 
         return !geometry::covered_by(p, g2, strategy);
     }
+
+    template <typename Strategy>
+    static inline bool apply(Geometry1 const& g1,
+                             Geometry2 const& g2,
+                             Strategy const& strategy,
+                             point1_type& p)
```

Comment:
Typically we put the output geometry before strategy.

Reply:
Btw, why is this type taken from Geometry1 and not passed as template parameter (passed by the user)?

Reply:
This is a limitation of the current design, i.e. closest_points call distance algorithm passing only the types of two geometries then distance calls parts of disjoint algorithm and at that point there is no user-defined type for the output point. However, that point will be an intersection point of two geometries. Do you think there could be a problem here?

Reply:
> Typically we put the output geometry before strategy.

I will fix this.

Reply:
> Do you think there could be a problem here?

Probably not. Technically when BG function returns a geometry the point type of this geometry should be used. But I think it's ok to convert to it at the very end. We should only use the more precise coordinate type of the two geometries though.